### PR TITLE
Contact update overhaul - again

### PR DESCRIPTION
**Pull request by annando**
Now some more data is fetched from the probing process. Additionally we make sure that duplicates don't occur and are defeated.

### CODE REVIEW

File: src/Model/GContact.php [hunk 1]
```diff
@@ -924,7 +924,7 @@ private static function updateFromPublicContact($condition)
 		}
 
 		// These fields are having different names but the same content
-		$gcontact['server_url'] = $contact['baseurl'];
+		$gcontact['server_url'] = defaults($contact, 'baseurl', ''); // "baseurl" can be null, "server_url" not
```
Comment (MrPetovan) [inline src/Model/GContact.php]: If `$contact['baseurl']` is guaranteed to exist at this point, please write `$contact['baseurl'] ?? ''` instead.

File: src/Network/Probe.php [hunk 2]
```diff
@@ -778,6 +783,12 @@ private static function pollNoscrape($noscrape_url, $data)
 			$data["poll"] = $json["dfrn-poll"];
 		}
 
+		if (isset($json["hide"])) {
+			$data["hide"] = (bool)$json["hide"];
```
Comment (MrPetovan) [inline src/Network/Probe.php]: Are you sure `hide` can't be the strings `"true"` or `"false"` ?

Comment (annando) [reply src/Network/Probe.php]: It is generated here: https://github.com/friendica/friendica/blob/develop/mod/noscrape.php#L47

Means: I'm totally sure that it isn't a string.

File: src/Model/GContact.php (hunk 1 above)
Comment (annando) [reply src/Model/GContact.php]: I really, really, really have to work myself into these structures. I always have to look at examples to understand it. What is it checking? ```$contact['baseurl']``` can be null, but does exist at any time.

Comment (MrPetovan) [reply src/Model/GContact.php]: `$var ?: $default` is `$var ? $var : $default`. However it doesn't work with non-existing variables/array keys.

`$var ?? $default` is `isset($var) ? $var : $default` so it doesn't complain for non-existing variables/array keys but it doesn't fall back to the default for falsy values.

File: src/Network/Probe.php (hunk 2 above)
Comment (MrPetovan) [reply src/Network/Probe.php]: The other problem is that the array key may not be existing. Please initialize it as `false` in the above `$json` array declaration.

Comment (annando) [reply src/Network/Probe.php]: I'm not sure where I should do this? In ```Probe.php``` I'm checking with ```isset```, so everything is fine, isn't it?

Comment (MrPetovan) [reply src/Network/Probe.php]: Ah, yeah, I forgot about that. Good point.

**Pull request merged**